test/integration/workspace/test_workspace.py

```diff
@@ -0,0 +1,264 @@
+                "sub/dep1/conanfile.py": GenConanfile("dep1", "0.1"),
+                "sub/dep2/conanfile.py": GenConanfile("dep2", "0.1")})
+        with c.chdir("sub"):
+            c.run("workspace add dep1")
+            assert "Reference 'dep1/0.1' added to workspace" in c.out
+            c.run("workspace add dep2")
+            assert "Reference 'dep2/0.1' added to workspace" in c.out
+            c.run("workspace info")
+            assert "dep1/0.1" in c.out
+            assert "dep2/0.1" in c.out
+        assert c.load_home("editable_packages.json") is None
+
+        c.run("editable list")
+        assert "dep1" not in c.out
+        assert "dep2" not in c.out
+        assert c.load_home("editable_packages.json") is None
+        with c.chdir("sub"):
+            c.run("editable add dep1")
+            assert c.load_home("editable_packages.json") is not None
+            c.run("editable list")
+            assert "dep1/0.1" in c.out
+            assert "dep2/0.1" not in c.out
+            c.run("workspace info")
+            assert "dep1" in c.out
+            assert "dep2" in c.out
+
+        c.run("editable list")
+        assert "dep1/0.1" in c.out
+        assert "dep2" not in c.out
+
+    @pytest.mark.parametrize("api", [False, True])
+    def test_dynamic_editables(self, api):
+        c = TestClient(light=True)
+        conanfile = textwrap.dedent("""
+            import os
+            from conan import ConanFile
+            from conan.tools.files import load
+            class Lib(ConanFile):
+                def set_name(self):
+                    self.name = load(self, os.path.join(self.recipe_folder, "name.txt"))
+                def set_version(self):
+                    self.version = load(self, os.path.join(self.recipe_folder, "version.txt"))
+            """)
+        if not api:
+            workspace = textwrap.dedent("""\
+                import os
+                name = "myws"
+
+                workspace_folder = os.path.dirname(os.path.abspath(__file__))
+
+                def editables():
+                    result = {}
+                    for f in os.listdir(workspace_folder):
+                        if os.path.isdir(f):
+                            name = open(os.path.join(f, "name.txt")).read().strip()
+                            version = open(os.path.join(f, "version.txt")).read().strip()
+                            result[f"{name}/{version}"] = {"path": f}
+                    return result
+                """)
+        else:
+            workspace = textwrap.dedent("""\
+               import os
+               name = "myws"
+
+               def editables(*args, **kwargs):
+                   result = {}
+                   for f in os.listdir(workspace_api.folder):
+                       if os.path.isdir(f):
+                           conanfile = workspace_api.load(os.path.join(f, "conanfile.py"))
+                           result[f"{conanfile.name}/{conanfile.version}"] = {"path": f}
```

Comment:
```suggestion
                           conanfile_path = os.path.join(f, "conanfile.py")
                           conanfile = workspace_api.load(conanfile_path)
                           result[f"{conanfile.name}/{conanfile.version}"] = {"path": conanfile_path}
```

found while playing around with the feature with @perseoGI

Reply:
Solved